lib/compat/wordpress-6.3/get-global-styles-and-settings.php

```diff
@@ -124,3 +124,37 @@ function wp_get_block_css_selector( $block_type, $target = 'root', $fallback = f
 function gutenberg_get_remote_theme_patterns() {
 	return WP_Theme_JSON_Resolver_Gutenberg::get_theme_data( array(), array( 'with_supports' => false ) )->get_patterns();
 }
+
+/**
+ * Gets the styles resulting of merging core, theme, and user data.
+ *
+ * @since 5.9.0
+ * @since 6.3.0 the internal link format "var:preset|color|secondary" is resolved
```

Comment:
Once introduced in WP Core, this modifies the value format for the returned styles from `wp_get_global_styles()`. As such, it is a backwards-compatibility (BC) break for extenders expecting and using this structure.

The result could mean broken styling experiences for users.

Is this a truly a bugfix, meaning `wp_get_global_styles()` should always have returned the style value structure? Or is this a new improvement to avoid leaking internal structures?

cc @oandregal

Reply:
This is why I think it's a fine change.

Let say we had the following input:

```php
"core/post-terms": {
    "typography": { "fontSize": "var(--wp--preset--font-size--small)" },
    "color":{ "background": "var:preset|color|secondary" }
}
```

**Before** this PR, the output of `wp_get_global_styles` would be:

```php
(
    [typography] => Array( [fontSize] => var(--wp--preset--font-size--small) )
    [color] => Array( [background] => var:preset|color|secondary )
)
```

**After** this PR, the output of `wp_get_global_styles` would be:

```php
(
    [typography] => Array( [fontSize] => var(--wp--preset--font-size--small) )
    [color] => Array( [background] => var(--wp--preset--color--secondary) )
)
```

Before this PR, the consumers had to deal with **two formats**: `var(--wp--preset--color--secondary)` and `var:preset|color|secondary`, as a possible representation of the same value. After this PR, they only have to deal with **one format** that's been always present: `var(--wp--preset--color--secondary)`. I don't remember when the `var:preset|color|secondary` was introduced for people to use in `theme.json` but it was later and added as a convenience. I did some quick search but didn't find when it was, I can do some archeology if this is important.

Note that the format doesn't carry any meaning about _where_ this data comes from. Any origin (default, block, theme, user) can use `var:preset|color|secondary`. Hence, consumers cannot use that format as a hint of it being "user data".

Does this help?